pymc/variational/opvi.py

```diff
@@ -854,12 +851,30 @@ def __init__(
         self.group = group
         self.user_params = params
         self._user_params = None
+        self.replacements = collections.OrderedDict()
+        self.value_replacements = collections.OrderedDict()
+        self.ordering = collections.OrderedDict()
         # save this stuff to use in __init_group__ later
         self._kwargs = kwargs
         if self.group is not None:
             # init can be delayed
             self.__init_group__(self.group)
 
+    def _prepare_start(self, start=None):
```

Comment:
Good idea to extract this, it will make it easier to add the type hints and switch to the new initval framework.